Seneca-CDOT/telescope
# Add build log to supabase

## Issue This PR Addresses
- Part of #2951
- I'm adding insert logic first before changing the way `status` gets build log


## Type of Change




## Description

### Help needed: 
- I'm not sure how to test it in staging. Local works

### Changes: 
- Add table build_logs to supabase
- Add route /build/:sha to retrieve build log
- Add route /build/add to add build log to supabase
- Add class BuildLog and storage.js to query data
- Add addBuildLog() in check-build.js 
- Expose SUPABASE_URL & SERVICE_ROLE_KEY to `status`



## Steps to test the PR
To test the feature locally, do the following:

- Change `autodeploymentUrl` in `src\api\status\public\js\build-log\check-build-status.js`

```js
const autodeploymentUrl = (path) => `//dev.telescope.cdot.systems/deploy${path}`;  
```

- Allow all `connectSrc` in `src\api\status\src\server.js`

```js
connectSrc: ["'self'", "*"]
```



## Checklist
